src/tag_gc.rs

```diff
@@ -3,30 +3,120 @@ use rustc_data_structures::fx::FxHashSet;
 use crate::*;
 
 pub trait VisitMachineValues {
-    fn visit_machine_values(&self, visit: &mut impl FnMut(&Operand<Provenance>));
+    fn visit_machine_values(&self, visit: &mut ProvenanceVisitor);
+}
+
+pub trait MachineValue {
```

Comment:
I don't understand the point of this trait. Why not to `impl VisitProvenance for Operand` etc?